dev/fattest.simplicity/src/componenttest/topology/impl/LibertyServer.java

```diff
@@ -2679,6 +2679,11 @@ protected void recursivelyCopyDirectory(RemoteFile remoteDirectory, LocalFile de
                 continue;
             }
 
+            if (remoteDirectory.getName().startsWith("dump_")) {
```

Comment:
I'm worried that this will skip out cases where we actually do want to copy folders that start with `dump_`, since this is added to a  generic method. It seems the root problem here is that `server.dumpServer()` returns while a server dump is not yet complete?

Reply:
@aguibert It's also an issue when the dump was done on a different thread than the test case (which is calling stop server and post stop server archive).  I considered using synchronization to prevent the server from stopping during a dump, but that seemed like overkill.  Do you prefer the synchronization?  Or have another suggestion?  

Thanks for the review comments

Reply:
Is there a particular reason why the server dump is done async? I think it would be better to turn `server.dumpServer()` into a blocking call. On the test client side we generally prefer to do things sequentially because it produces more reliable/repeatable tests and is easier to debug.

Reply:
side note: doing a server dump is a very resource intensive process, and generally we try to avoid doing it in the main path because it consumes so much CPU and disk space. Unless you are explicitly testing the contents of a server dump here, I'd recommend you remove the server dump call, or at least turn it into a failure-only code path (although in general if a server hangs the test framework will take a server dump automatically)

Reply:
This was made to be async so that server dumps could be acquired at various phases of the test at runtime.  It enables users to get thread dump info, as well as other introspection data at various times throughout the test.  The "unpredictable" timing of the dumps is ok in most cases because it is only for diagnosing tests that are already failing.  Ideally, we would get the data from the dumps in all cases, but what we have now are tests failing because we are collecting the dumps - that is no good.